src/gpu-compute/compute_unit.cc

```diff
@@ -1042,6 +1069,28 @@ ComputeUnit::SQCPort::recvReqRetry()
     }
 }
 
+const char*
+ComputeUnit::SQCPort::MemReqEvent::description() const
+{
+    return "ComputeUnit SQC memory request event";
+}
+
+void
+ComputeUnit::SQCPort::MemReqEvent::process()
+{
+    SenderState *sender_state = safe_cast<SenderState*>(pkt->senderState);
+    [[maybe_unused]] ComputeUnit *compute_unit = sqcPort.computeUnit;
+
+    if (pkt->req->systemReq()) {
+        assert(compute_unit->shader->systemHub);
+        SystemHubEvent *resp_event = new SystemHubEvent(pkt, &sqcPort);
+        compute_unit->shader->systemHub->sendRequest(pkt, resp_event);
```

Comment:
I don't think this path is possible.  If it is, since the SystemHubEvent process() does nothing we will have requests silently dropped.  I would instead make this an assert that it is not a system request.  If we see the assert in the future we'll know exactly where to go back and fix it

Reply:
Thanks for letting me know. I was not sure about this part earlier. I have now changed this to an assert and the code still works